File: VB/EF/ExtendAuditEF/ExtendAuditEF.Module/BusinessObjects/CustomAuditDataItem.vb
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.BaseImpl.EFCore.AuditTrail

Namespace dxTestSolution.Module.BusinessObjects

    <DomainComponent>
    Public Class CustomAuditDataItem
        Inherits NonPersistentBaseObject

        Private sourceAuditDataItem As AuditDataItemPersistent

        Private targetObjectNameField As String

        Public Sub New(ByVal sourceAuditDataItem As AuditDataItemPersistent, ByVal targetObjectName As String)
            Me.sourceAuditDataItem = sourceAuditDataItem
            targetObjectNameField = targetObjectName
        End Sub

        Public ReadOnly Property TargetObjectName As String
            Get
                Return targetObjectNameField
            End Get
        End Property

        Public ReadOnly Property Description As String
            Get
                Return sourceAuditDataItem.Description
            End Get
        End Property

        Public ReadOnly Property ModifiedOn As Date
            Get
                Return sourceAuditDataItem.ModifiedOn
            End Get
        End Property

        Public ReadOnly Property NewValue As String
            Get
                Return sourceAuditDataItem.NewValue
            End Get
        End Property

        Public ReadOnly Property OldValue As String
            Get
                Return sourceAuditDataItem.OldValue
            End Get
        End Property

        Public ReadOnly Property OperationType As String
            Get
                Return sourceAuditDataItem.OperationType
            End Get
        End Property

        Public ReadOnly Property PropertyName As String
            Get
                Return sourceAuditDataItem.PropertyName
            End Get
        End Property

        Public ReadOnly Property UserName As String
            Get
                Return sourceAuditDataItem.UserName
            End Get
        End Property
    End Class
End Namespace

File: CS/EF/ExtendAuditEF/ExtendAuditEF.Module/BusinessObjects/CustomAuditDataItem.cs
using System;
using System.Collections.Generic;
using System.Text;

using DevExpress.ExpressApp.DC;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp;
using DevExpress.Persistent.BaseImpl.EFCore.AuditTrail;

namespace dxTestSolution.Module.BusinessObjects {
    [ DomainComponent]
    public class CustomAuditDataItem: NonPersistentBaseObject {
        private AuditDataItemPersistent sourceAuditDataItem;
        private string targetObjectName;
        public CustomAuditDataItem(AuditDataItemPersistent sourceAuditDataItem, string targetObjectName) {
            this.sourceAuditDataItem = sourceAuditDataItem;
            this.targetObjectName = targetObjectName;
        }
        public string TargetObjectName { get { return targetObjectName; } }
        public string Description { get { return sourceAuditDataItem.Description; } }
        public DateTime ModifiedOn { get { return sourceAuditDataItem.ModifiedOn; } }
        public string NewValue { get { return sourceAuditDataItem.NewValue; } }
        public string OldValue { get { return sourceAuditDataItem.OldValue; } }
        public string OperationType { get { return sourceAuditDataItem.OperationType; } }
        public string PropertyName { get { return sourceAuditDataItem.PropertyName; } }
        public string UserName { get { return sourceAuditDataItem.UserName; } }
    }
}
